build-tools/src/main/resources/software/amazon/awssdk/checkstyle-suppressions.xml

```diff
@@ -42,4 +42,8 @@
     <!-- Assumes getter/setter match JSON property -->
     <suppress checks="AbbreviationAsWordInName"
               files=".*[\\/]software[\\/]amazon[\\/]awssdk[\\/]services[\\/]s3[\\/]event[\\/]S3EventNotification.java$"/>
+
+    <!-- This check is problematic with lambda. https://github.com/checkstyle/checkstyle/issues/3342 -->
+    <suppress checks="Indentation"
+              files="AsyncApiCallTimeoutTrackingStage.java|MakeAsyncHttpRequestStage.java"/>
```

Comment:
Can we just work around it instead? Turning off indentation checking in general is not great, and we've worked around it elsewhere.

Reply:
I tried but I find it is very ugly to comply with the checkstyle rules when you have lambda as parameter.

```
        return TimerUtils.timeAsyncTaskIfNeeded(executeFuture,
                                                timeoutExecutor, 
           () -> ApiCallAttemptTimeoutException.create(timeoutMillis),
                                                timeoutMillis);
```

Reply:
Can you factor it into a separate method, assign it to a variable, or wrap after the `->`?

Reply:
Yeah, creating variable would work but I was not sure if it was worth it to just comply with checkstyle rules. If we think keeping indention checkstyle is important enough, I can make the change.

Reply:
If we could have a more targeted checkstyle rule, I'd favor disabling it, but the indentation checkstyle catches a lot of other issues.

Reply:
Yeah, updated